ext/app/electron/config.ts

```diff
@@ -86,6 +86,11 @@ export async function loadConfig() {
     NO_VALIDATION,
     path.join(electron.app.getPath("home"), ".grist")
   );
+  validateOrFallback(
+    "PYTHON_VERSION_ON_CREATION",
+    NO_VALIDATION,
```

Comment:
At first glance, is it a 2 or 3 choice? If so, could it be better to always fallback to 3 if not set explicitly to `2`?

Reply:
Yes, 2 or 3 are the choices. It should always be "3". There's almost no reason to create a new document with Python 2.

However, upon discussing this with @paulfitz, we thought maybe a deeper fix should be needed. In grist-core, if the document doesn't specify its Python version (should be in the `_grist_DocInfo` table), *and* the Grist instance doesn't specify a `PYTHON_VERSION`, then Grist defaults to Python 2. This is how users were experiencing this problem, because Desktop doesn't set the engine version in the document. Thus, exporting Desktop documents and importing them into SaaS was resulting in broken formulas.

The Python version should probably default to 3 instead if neither the document nor the Grist instance specifies what to default to.

Reply:
Thanks for the context!

What concerns me is the `NO_VALIDATION`. If you write a simple validator `(ver) => ["2", "3"].includes(ver)`, `loadConfig()` will default to 3 when it sees an otherwise invalid value (something other than 2 and 3). I think this is likely desired, even only as a temporary precaution.

Of course, I agree that grist-core itself should always default to 3.

Reply:
Oh, sure, I've added a validator.